r/R/type.R

```diff
@@ -68,17 +69,50 @@ FLOAT_TYPES <- c("float16", "float32", "float64", "halffloat", "float", "double"
 #' type(c("A", "B", "C"))
 #' type(mtcars)
 #' type(Sys.Date())
+#' type(as.POSIXlt(Sys.Date()))
+#' type(vctrs::new_vctr(1:5, class = "my_custom_vctr_class"))
 #' @export
-type <- function(x) UseMethod("type")
+type <- function(x) infer_type(x)
```

Comment:
This is probably safe to do but you should add to deprecated.R something like

```
type <- function(x) {
  .Deprecated("infer_type")
  infer_type(x)
}
```

in case anyone was using this.